source/Halibut/Transport/Streams/StreamExtensionMethods.cs

```diff
@@ -16,5 +17,20 @@ public static async Task WriteControlLineAsync(this Stream stream, string s, Can
             var bytes = Encoding.GetBytes(s + ControlMessageNewLine);
             await stream.WriteAsync(bytes, 0, bytes.Length, cancellationToken);
         }
+        
+        public static async Task WriteLineAsync(this Stream stream, string s, CancellationToken cancellationToken)
+        {
+            var bytes = Encoding.GetBytes(s + ControlMessageNewLine);
```

Comment:
nit: ControlMessageNewLine doesn't make sense here.

The reason for having Control* stuff defined separately as this is an area that can never change e.g. for backward compatibility, we can never change the behaviour.  The thought was to leave these alone and not use them so it's clear that there is a specific contract for how control messages are sent over the stream.

Reply:
Similar to the other nit, I flip-flopped on whether to use `Environment.NewLine` here instead, for the exact reasons you've stated. Ultimately I decided to leave it as I thought purposely _not using_ `ControlMessageNewLine` here,  instead replacing with `Environment.NewLine`, might be confusing. Too much second-guessing! 😵‍💫 

I will update.

Reply:
`Environment.NewLine` will change the behaviour depending on the OS hosting the Halibut Runtime. I would suggest maintaining the behaviour of the sync code that uses `using (var writer = new StreamWriter(stream, new UTF8Encoding(false)) { NewLine = "\r\n" })`